src/utils/access.ts

```diff
@@ -0,0 +1,27 @@
+import { ROUTES } from '@/sources/routes';
+
+export type AccessPatterns = {
+  protected: string[];
+  public: string[];
+};
+
+export function buildAccessPatterns(): AccessPatterns {
+  return {
+    protected: [ROUTES.history],
+    public: [ROUTES.login, ROUTES.signup],
+  };
+}
+
+export function isProtectedPath(
+  pathnameNoLocale: string,
+  patterns: AccessPatterns
+): boolean {
+  return patterns.protected.includes(pathnameNoLocale);
+}
+
+export function isPublicPath(
+  pathnameNoLocale: string,
+  patterns: AccessPatterns
+): boolean {
+  return patterns.public.includes(pathnameNoLocale);
+}
```

Comment:
а вот здесь можно вот так, по суть в мидлваре же нужны только проверки на принадлежность к защищенным/открытым

```suggestion
export const accessPatterns: AccessPatterns = {
  protected: [ROUTES.history],
  public: [ROUTES.login, ROUTES.signup],
};

export function isProtectedPath(pathnameNoLocale: string): boolean {
  return accessPatterns.protected.includes(pathnameNoLocale);
}

export function isPublicPath(pathnameNoLocale: string): boolean {
  return accessPatterns.public.includes(pathnameNoLocale);
}
```